### ISSUE TITLE
Form Parameters request matching is broken

### ISSUE THREAD

**Issue by veskoy** [wiremock/wiremock#2363]
### Proposal

The [form parameters request matching](https://wiremock.org/docs/request-matching/#request-with-form-parameters) doesn't work with the most up-to-date version of WireMock 3.0.2.

After I went through the stack, it looks like the original `HttpServletRequest` contains the form parameters, but they get lost when the request is wrapped in `WireMockHttpServletRequestAdapter` here:
https://github.com/wiremock/wiremock/blob/b3659fb7b86bcd66b8f8e50afad5190b2922aade/src/main/java/com/github/tomakehurst/wiremock/servlet/WireMockHandlerDispatchingServlet.java#L145-L147

The cause for the issue seems to be the `getFormParameters` method of the `WireMockHttpServletRequestAdapter` class which is strangely trying to obtain the form parameters from the request body, rather than the form parameters property:
https://github.com/wiremock/wiremock/blob/b3659fb7b86bcd66b8f8e50afad5190b2922aade/src/main/java/com/github/tomakehurst/wiremock/servlet/WireMockHttpServletRequestAdapter.java#L320-L337

After I checked the code history, it looks like the `getFormParameters` method was properly implemented by @kapishmalik in #2157 to utilise the `HttpServletRequest`'s `getParameterMap`:
https://github.com/wiremock/wiremock/blob/b3659fb7b86bcd66b8f8e50afad5190b2922aade/src/main/java/com/github/tomakehurst/wiremock/servlet/WireMockHttpServletRequestAdapter.java#L320-L337

However it got broken with the following PR #2238 @tomakehurst  the implementation was maybe unintentionally reverted to the old behaviour which depends on the request body:
https://github.com/wiremock/wiremock/pull/2238/files#diff-754490dca33832006201df079fb169190351e7368bab1f83c5a3a48a8ef841f4R330-R347

This issue also affects the multipart form data matching, which is also broken.

### Reproduction steps

The issue is reproducible with the form parameters example provided in the documentation:
https://wiremock.org/docs/request-matching/#request-with-form-parameters

**Comment by tomakehurst** [comment]
Please can you share your reproduction steps?

The form JSON example in the docs is malformed so I'll publish a fix for that, but when I tested it with a well-formed version of that example it worked fine.

**Comment by veskoy** [comment]
> Please can you share your reproduction steps?
> 
> The form JSON example in the docs is malformed so I'll publish a fix for that, but when I tested it with a well-formed version of that example it worked fine.

Something I forgot to mention is that I am using WireMock embedded into a servlet container. I suppose the WireMock Standalone doesn't use the `WireMockHttpServletRequestAdapter` class, so maybe that's why you are not reproducing the issue.

Also I'm not using JSON to setup the WireMock server - I'm using the DSL as follows:
```java
 stubFor(post(urlPathEqualTo("/form"))
       .withFormParam("tool", equalTo("WireMock"))
       .willReturn(aResponse()));
```

Here are some debugging screenshots which might help you:
![image](https://github.com/wiremock/wiremock/assets/5436204/22d94fd2-76d3-44c8-8e12-dc52207a616e)
![image](https://github.com/wiremock/wiremock/assets/5436204/6fd329c2-7f0c-4422-aac7-ff7416f90266)
![image](https://github.com/wiremock/wiremock/assets/5436204/ff036eb9-ad4b-4c2e-b712-325d91908f10)

**Comment by veskoy** [comment]
It turned the issue was caused by a request wrapper class which didn't add the form parameters to request body, but added them to the formParameters map only. I managed to have it working successfully by tweaking the wrapper class a bit.

I still believe that it would be better if `com.github.tomakehurst.wiremock.servlet.WireMockHttpServletRequestAdapter#getFormParameters` method utilises `jakarta.sevlet.ServletRequest#getParameterMap` instead of parsing the request body input stream.
